Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -520,7 +520,15 @@ public final class PackageBuilder {
                 if subpath == "" || subpath == "." {
                     return packagePath
                 }
-                let path = packagePath.appending(RelativePath(subpath))
+
+                let path: AbsolutePath
+
+                if subpath.hasPrefix("/") {
+                    path = AbsolutePath(subpath)
+                } else {
+                    path = AbsolutePath(subpath, relativeTo: packagePath)
+                }
```

Comment:
I was thinking an API like this will be useful here (and maybe some other places):
```
enum PathType {
     case absolute
     case relative
     case invalid
}

public static func check(string: String) -> PathType
```